ixmp/backend/jdbc.py

```diff
@@ -405,7 +433,15 @@ def get(self, ts, version):
         try:
             jobj = method(*args)
         except java.IxException as e:
-            raise RuntimeError(*e.args) from None
+            # Try to re-raise as a ValueError for bad model or scenario name
+            match = re.search(r"table '([^']*)' from the database", e.args[0])
```

Comment:
maybe better to change exception handling on java side to generate proper error message?

Reply:
Longer-term, I think it is best to specify on the Python side, i.e. through "Raises" sections in the base.Backend docstrings, and through tests, what happens for each method if something goes wrong.

When touching existing code, I think, "What about a hypothetical Backend that's *not* backed by a database?" Ideally, the same exceptions—type and message—should be raised by this hypothetical Backend and by JDBCBackend for the same user error. This means the user can write `try:`/`except…` code that works no matter which Backend is used.

PRs are welcome that change things on the Java side and remove this catch-and-reraise code.

Reply:
What I meant is - to avoid relying on specific exception messages. If needed to distinguish the cause we may (a) introduce error codes or (b) create more exception types in java (e.g. IxNotFoundException or similar).
Relying on exception message is often fragile.

Reply:
Separate PRs would also be welcome to do that!

Reply:
Is it ok if we create separate issue? I had a quick look on existing code and it seems to be common pattern in many places to parse error messages.

Reply:
Created #296.